plugins/out_stackdriver/stackdriver.c

```diff
@@ -2625,62 +2831,87 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
     if (ret != 0) {
         flb_plg_warn(ctx->ins, "http_do=%i", ret);
         ret_code = FLB_RETRY;
-#ifdef FLB_HAVE_METRICS
-        update_http_metrics(ctx, event_chunk, ts, STACKDRIVER_NET_ERROR);
-#endif
     }
     else {
         /* The request was issued successfully, validate the 'error' field */
         flb_plg_debug(ctx->ins, "HTTP Status=%i", c->resp.status);
         if (c->resp.status == 200) {
             ret_code = FLB_OK;
         }
-        else if (c->resp.status >= 400 && c->resp.status < 500) {
-            ret_code = FLB_ERROR;
-            flb_plg_warn(ctx->ins, "error\n%s",
-                c->resp.payload);
-        }
         else {
+          /* check partial success */
+          ret_partial_success =
+                parse_partial_success_response(c,
```

Comment:
Do we need to do this if there aren't metrics? Should this be inside the ifdef below?